spark-plugins/widgets/KinesisSource-streamingsource.json

```diff
@@ -0,0 +1,108 @@
+{
+  "metadata": {
+    "spec-version": "1.0"
+  },
+  "configuration-groups": [
+    {
+      "label": "Kinesis Configuration",
+      "properties": [
+        {
+          "widget-type": "textbox",
+          "label": "Reference Name",
+          "name": "referenceName"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Application Name",
+          "name": "appName"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Stream Name",
+          "name": "streamName"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Kinesis endpoint url",
+          "name": "endpointUrl"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "region",
+          "name": "region"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Checkpoint interval duration",
+          "name": "duration"
+        },
+        {
+          "widget-type": "select",
+          "label": "Initial position in stream",
+          "name": "initialPosition",
+          "widget-attributes": {
+            "values": [
+              "LATEST",
+              "TRIM_HORIZON"
+            ],
+            "default": "LATEST"
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "AWS access key id",
```

Comment:
Not sure how the UI works but if you are saying they will be stored in the secure store then these should not be textbox right ? There is some other widget for secure info